### PR TITLE
Add AppDomain.CurrentDomain event asserts

### PR DESCRIPTION
**Pull request by sdmaclea**
Adds tests for dotnet/coreclr#25019

Blocked on bits from dotnet/coreclr#25019 making it to corefx

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 1]
```diff
@@ -519,9 +519,13 @@ public void AssemblyLoad()
                 bool AssemblyLoadFlag = false;
                 AssemblyLoadEventHandler handler = (sender, args) =>
                 {
+                    Assert.Equal(AppDomain.CurrentDomain, sender);
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Assert.Same?

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 2]
```diff
@@ -541,14 +545,18 @@ public void AssemblyLoad()
...
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Does not support Assembly.LoadFile")]
         public void AssemblyResolve()
         {
             CopyTestAssemblies();
 
             RemoteExecutor.Invoke(() => {
-                ResolveEventHandler handler = (sender, e) =>
+                ResolveEventHandler handler = (sender, args) =>
                 {
+                    Assert.Equal(AppDomain.CurrentDomain, sender);
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Ditto

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 3]
```diff
@@ -561,17 +569,20 @@ public void AssemblyResolve()
...
             CopyTestAssemblies();
 
             RemoteExecutor.Invoke(() => {
                 Assembly a = Assembly.LoadFile(Path.Combine(Environment.CurrentDirectory, "TestAppOutsideOfTPA.exe"));
 
-                ResolveEventHandler handler = (sender, e) =>
+                ResolveEventHandler handler = (sender, args) =>
                 {
-                    Assert.Equal(e.RequestingAssembly, a);
+                    Assert.Equal(AppDomain.CurrentDomain, sender);
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Same for all of these; will stop commenting.

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 4]
```diff
@@ -561,17 +569,20 @@ public void AssemblyResolve()
...
+                    Assert.NotNull(args);
+                    Assert.NotNull(args.Name);
+                    Assert.Equal(args.RequestingAssembly, a);
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: And these, presumably

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs (hunk 1 above)
Comment (sdmaclea) [reply src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: You think I would remember from the last time you told me 😄

File: src/System.Runtime.Extensions/tests/System/AppDomainTests.cs [hunk 5]
```diff
@@ -541,37 +546,47 @@ public void AssemblyLoad()
...
+                    Assert.NotNull(args.RequestingAssembly);
+                    AssemblyResolveFlag = true;
                     return Assembly.LoadFile(Path.Combine(Environment.CurrentDirectory, "AssemblyResolveTestApp.dll"));
                 };
 
                 AppDomain.CurrentDomain.AssemblyResolve += handler;
 
                 Type t = Type.GetType("AssemblyResolveTestApp.Class1, AssemblyResolveTestApp", true);
                 Assert.NotNull(t);
+                Assert.True(AssemblyResolveFlag);
```
Comment (sdmaclea) [inline src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: I was surprised this didn't fail in the first run.  Added this extra check to be sure the handler is run....

Comment (sdmaclea) [reply src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: This test is not running the handler...

Comment (sdmaclea) [reply src/System.Runtime.Extensions/tests/System/AppDomainTests.cs]: Added a test which will run the handler and created issue #38361

**Pull request merged**